.github/workflows/codeql-analysis.yml

```diff
@@ -27,6 +27,8 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
The issue was hard to track down but eventually isolated it to the lack of history broke the local version calculation and led to the local build being mismatched with the locks (if the bom contains self references, since we recently added attribute service to bom).